messaging/src/main/java/org/axonframework/messaging/deadletter/DeadLetter.java

```diff
@@ -0,0 +1,106 @@
+ * evaluation once the {@link #expiresAt()} time is reached. Upon successful evaluation the letter can be cleared
+ * through {@link #acknowledge()}. The {@link #requeue()} method should be used to signal evaluation failed, reentering
+ * the letter into its queue.
+ *
+ * @param <T> The type of {@link Message} represented by this interface.
+ * @author Steven van Beelen
+ * @since 4.6.0
+ */
+public interface DeadLetter<T extends Message<?>> {
+
+    /**
+     * The identifier of this dead-letter.
+     *
+     * @return The identifier of this dead-letter.
+     */
+    String identifier();
+
+    /**
+     * The {@link QueueIdentifier} this dead-letter belongs to.
+     *
+     * @return The {@link QueueIdentifier} this dead-letter belongs to
+     */
+    QueueIdentifier queueIdentifier();
+
+    /**
+     * The {@link Message} of type {@code T} contained in this letter.
+     *
+     * @return The {@link Message} of type {@code T} contained in this letter.
+     */
+    T message();
+
+    /**
+     * The {@link Cause cause} for the {@link #message()} to be dead-lettered. May be {@code null} in case this letter
+     * is enqueued as a consequence of earlier letters with the same {@link QueueIdentifier}.
+     *
+     * @return The {@link Cause cause} for the {@link #message()} to be dead-lettered
+     */
+    @Nullable
+    Cause cause();
+
+    /**
+     * The moment in time when the {@link #message()} was dead-lettered.
+     *
+     * @return The moment in time when the {@link #message()} was dead-lettered.
+     */
+    Instant deadLettered();
+
+    /**
+     * The moment in time when this letter will expire. Should be used to deduce whether the letter is ready to be
+     * evaluated. Will equal the {@link #deadLettered()} value if this letter is enqueued as part of a sequence to
+     * ensure it is available right after a previous letter within the same sequence.
+     *
+     * @return The moment in time when this letter will expire.
+     */
+    Instant expiresAt();
+
+    /**
+     * The number of retries this {@link DeadLetter dead-letter} has gone through.
+     *
+     * @return The number of retries this {@link DeadLetter dead-letter} has gone through.
+     */
+    int numberOfRetries();
+
+    /**
+     * Acknowledges this {@link DeadLetter dead-letter} as successfully evaluated. This operation will remove this
+     * letter from its queue.
+     */
+    void acknowledge();
+
+    /**
+     * Reenters this {@link DeadLetter dead-letter} in the queue it originates from. This method should be used to
+     * signal the evaluation failed.
+     * <p>
+     * The operation will adjust the {@link #expiresAt()} to the current time and increment the
+     * {@link #numberOfRetries()}. This operation might remove this letter from the {@link DeadLetterQueue queue} it
+     * originated from.
+     */
+    void requeue();
```

Comment:
Should the `requeue` operation also schedule the evaluation task again? Strictly speaking the `DeadLetterQueue` implementation sets the operation, but drafting this in the JavaDoc, if we take this road, is important.
Thoughts?

Reply:
It's not the responsibility of the message to know what happens when it is requeued, it's the responsibility of the DLQ implementation. 
From an OO-programming perspective, `DeadLetter.requeue` makes sense. However, since it has no logic except for forwarding to the DLQ implementation, I think `DeadLetterQueue.requeue(DeadLetter) makes **more** sense.

Reply:
The downside of _that_ approach is that the `DeadLetterQueue` will need implementation-specific details of the `DeadLetter`, which means that although the API will accept a `DeadLetter`, the implementation will need to cast that to the specific type used by the `DeadLetterQueue`. By putting the method on the `DeadLetter`, you fully encapsulate this behavior.

Reply:
Resolving this comment as the `DeadLetter#requeue` method is removed in favor of a `DeadLetterQueue#requeue` method. This indeed means the `DeadLetter` is constructed outside of the Dead Letter Queue, but doing so let's the queue align closer with the general concept of a dead-letter queue.